File: May 2021/May_04.java
class Solution {
    public boolean checkPossibility(int[] nums) {
        int count = 0;
        for(int i = 1; i < nums.length; i++){
            if(nums[i-1] > nums[i]){
                count++;
                if(i-2 >= 0 && i+1 < nums.length){
                    if(nums[i-2] > nums[i] && nums[i-1] > nums[i+1]) return false;
                }
                if(count > 1) return false;
            }
        }
        return true;
    }
}

File: May 2022/May_04.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int i = 0, j = nums.size() - 1, count = 0;
        while(i < j){
            if(nums[i] + nums[j] > k)
                j--;
            else if(nums[i] + nums[j] < k)
                i++;
            else{
                i++;
                j--;
                count++;
            }
        }
        return count;
    }
};
